container_files/assets/www/doc/builtin/functions/Tfidf.md

```diff
@@ -7,8 +7,39 @@ appears in a reference corpus.
 ## Configuration
 ![](%%config function tfidf)
 
+### Type of TF scoring
+
+Given a term \\( t \\), a document \\( d \in D \\) and the term frequency denoted by \\(f_{t,d}\\):
+
+| *weighting scheme* | *description* | *TF weight* |
+|--------------------|---------------|-------------|
+| raw | the term frequency in the document | \\(f_{t,d}\\) |
+| log | the logarithm of the term frequency in the document | \\( \log(1 + f_{t,d}) \\) |
+| augmented | the half term frequency in the document divided by the maximum frequency of any term in the document | \\( 0.5 + 0.5 \cdot \frac { f_{t,d} }{\max_{\{t' \in d\}} {f_{t',d}}} \\) |
+
+### Type of IDF scoring
+
+Given:
+
+- \\( N \\): the total number of documents in the corpus (\\( N = |D| \\))
+- \\( n_t = |\{d \in D: t \in d\}| \\): number of documents where the term \\( t \\) appears in
+
+| *weighting scheme* | *description* | *IDF weight*  |
+|--------------------|---------------|--------------|
+| unary | unary IDF score, i.e. don't use IDF | \\( 1 \\) |
+| inverse | the logarithm of the number of document in the corpus divided by the number of documents the term appears in | \\( \log \left( \frac {1 + N}{1 + n_t } \right) \\) |
+| inverseSmooth | similar to inverse but with logarithmic terms above 1 | \\( \log \left( 1 + \frac {1 + N}{1 + n_t } \right) \\) |
```

Comment:
The _inverse_ is already smoothed to prevent division by zero.  May be here, it would help our understanding if the description mentions why one would want this type of IDF instead of describing the formula to achieve it.  I guess a distinctive feature of _inverseSmooth_ is that it lowers the score of rare terms compare to more frequent terms.